src/message.rs

```diff
@@ -0,0 +1,161 @@
+use anyhow::{anyhow, Error};
```

Comment:
Crate/libraries should be using the `thiserror` crate, apps and users of the library should have the ability to recover from errors. The crate `anyhow` is intended for applications where recovering from errors may not be an option.